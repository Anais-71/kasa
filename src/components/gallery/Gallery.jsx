import React, { useState, useEffect } from 'react'
import { useParams } from 'react-router-dom'
import { fetchData } from '../../utils/fetch'
import { FaChevronLeft, FaChevronRight } from 'react-icons/fa'

//CSS
import './gallery.css'

function Gallery() {
  const { id } = useParams()
  const [images, setImages] = useState([])
  const [currentIndex, setCurrentIndex] = useState(0)

  useEffect(() => {
    const getData = async () => {
      const dataJson = await fetchData()
      const dataItem = dataJson.find((item) => item.id === id)
      setImages(dataItem ? dataItem.pictures : [])
    }
    getData()
  }, [id])

  const goNext = () => {
    setCurrentIndex((currentIndex + 1) % images.length)
  }

  const goPrev = () => {
    setCurrentIndex((currentIndex - 1 + images.length) % images.length)
  }

  if (images.length === 0) {
    return null
  }

  return (
    <div className="gallery">
      {images.length > 1 && (
        <FaChevronLeft className="gallery__prev" onClick={goPrev} />
      )}
      <img
        className="gallery__img"
        src={images.length > 0 ? images[currentIndex] : 'default.jpg'}
        alt="Logement"
      />
      {images.length > 1 && (
        <FaChevronRight className="gallery__next" onClick={goNext} />
      )}
      <div className="gallery__counter">
        {currentIndex + 1}/{images.length}
      </div>
    </div>
  )
}

export default Gallery
