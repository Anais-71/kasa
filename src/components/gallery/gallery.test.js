import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react';
import Gallery from './Gallery';
import { fetchData } from '../../utils/fetch';
import { BrowserRouter as Router } from 'react-router-dom';

jest.mock('../../utils/fetch');

const mockData = [
    {
        id: '1',
        pictures: ['image1.jpg', 'image2.jpg', 'image3.jpg']
    },
    {
        id: '2',
        pictures: ['image4.jpg', 'image5.jpg', 'image6.jpg']
    }
];

describe('When I am on a housing page', () => {
    beforeEach(() => {
        fetchData.mockResolvedValue(mockData);
    });
    
    it('Then the image changes when next button is clicked', async () => {
        const { findByAltText, container } = render(<Router><Gallery /></Router>);
        await waitFor(() => expect(fetchData).toHaveBeenCalledTimes(1));

        const nextButton = container.querySelector('.gallery__next');
        if (nextButton) {
            fireEvent.click(nextButton);
            const image = await findByAltText('Logement');
            expect(image.src).toContain('image2.jpg');
        } else {
            console.log('Next button is not available');
        }
    });

    it('Then the image changes when previous button is clicked', async () => {
        const { findByAltText, container } = render(<Router><Gallery /></Router>);
        await waitFor(() => expect(fetchData).toHaveBeenCalledTimes(1));

        const prevButton = container.querySelector('.gallery__prev');
        if (prevButton) {
            fireEvent.click(prevButton);
            const image = await findByAltText('Logement');
            expect(image.src).toContain('image3.jpg');
        } else {
            console.log('Prev button is not available');
        }
    });

    test('The gallery component matches the snapshot', () => {
        const { asFragment } = render(<Router><Gallery /></Router>);
        expect(asFragment()).toMatchSnapshot();
    }); 
});