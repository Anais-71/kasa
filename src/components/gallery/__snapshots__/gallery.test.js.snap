// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When I am on a housing page The gallery component matches the snapshot 1`] = `<DocumentFragment />`;
